regxmllib/CMakeLists.txt

```diff
@@ -1,29 +1,92 @@
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.6)
 project (regxmllibc)
+option(BUILD_SHARED_LIB Bool OFF)
+
+# Set build type
+if(BUILD_SHARED_LIB)
+set(LIB_TYPE SHARED)
+if(WIN32 AND NOT CYGWIN)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+endif(WIN32 AND NOT CYGWIN)
+else(BUILD_SHARED_LIB)
+set(LIB_TYPE STATIC)
+add_definitions( -DUTILS_LIB_STATIC )
+endif(BUILD_SHARED_LIB)
+message(STATUS "*** LIB_TYPE is ${LIB_TYPE}")
 
 include(FindXercesC)
 find_package(XercesC REQUIRED)
 include_directories( ${XercesC_INCLUDE_DIR}
 						src/main/cpp
 					)
 
+# Append "_d" if debug lib.
+set(CMAKE_DEBUG_POSTFIX _d) 
 
-file(GLOB_RECURSE SRC_FILES src/main/cpp/*.cpp src/main/cpp/*.h )
-add_library(${PROJECT_NAME} ${SRC_FILES})
-install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 98)
-
+file(GLOB_RECURSE SRC_FILES src/main/cpp/*.cpp )
+file(GLOB_RECURSE INC_FILES src/main/cpp/*.h )
+add_library(${PROJECT_NAME} ${LIB_TYPE} ${SRC_FILES} ${INC_FILES})
 
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 98)
 
 target_link_libraries ( ${PROJECT_NAME} ${XercesC_LIBRARY} )
 
-foreach(source IN LISTS SRC_FILES)
+#  Set Visual Studio filters for nested sourcce paths
+foreach(source IN LISTS SRC_FILES INC_FILES)
 	file(RELATIVE_PATH rel_source ${CMAKE_CURRENT_SOURCE_DIR}/src/main/cpp/com/ ${source})
     get_filename_component(source_path ${rel_source} DIRECTORY)
     string(REPLACE "\\" "/" source_path_msvc "${source_path}")
     source_group("${source_path_msvc}" FILES "${source}")
 endforeach()
 
+if(WIN32 AND NOT CYGWIN)
+  set(DEF_INSTALL_CMAKE_DIR CMake)
```

Comment:
Shouldn't this be set by the user?

Reply:
The resulting INSTALL_CMAKE_DIR can be edited by the user after configuring the build and before generating the build files. DEF_INSTALL_CMAKE_DIR holds useful default locations.